src/uwtools/config/core.py

```diff
@@ -332,31 +332,14 @@ def parse_include(self, ref_dict: Optional[dict] = None) -> None:
                     self.update_values(self._load_paths(filepaths))
                     del ref_dict[key]
 
-    def str_to_type(self, s: str) -> Union[bool, float, int, str]:
+    def reify_str(self, s: str) -> Union[bool, float, int, str]:
         """
-        Reify a string to a Python object, if possible.
-
-        Try to convert Boolean-ish values to bool objects, int-ish values to ints objects, and
-        float-ish values to float objects. Return the original string if none of these apply.
+        Reify a string to a Python object, using YAML. Jinja2 templates will be passed as-is.
 
         :param s: The string to reify.
         """
         s = s.strip("\"'")
-        if s.lower() in ["true", "yes", "yeah"]:
-            return True
-        if s.lower() in ["false", "no", "nope"]:
-            return False
-        # int
-        try:
-            return int(s)
-        except ValueError:
-            pass
-        # float
-        try:
-            return float(s)
-        except ValueError:
-            pass
-        return s
+        return s if "{{" in s or "{%" in s else yaml.safe_load(s)
```

Comment:
1. I love the simplicity of this. I think my description in Jira suggested something more complicated, like repeatedly asking `yaml` to parse the value and seeing when it changed, but now I'm not sure why. This seems right: Either `yaml` parses the `str` to a new value and returns it, or it remains a `str` and it returns it. All good, right?
2. But what about an `s` like `{ n: 88 }` or `[1, 2, 3]`? Those will get parsed into `dict` and `list` objects, and those aren't part of our return type, so should we just return `s` if we get one of these back from `safe_load()`? Or should we raise an exception here since we got something we didn't expect? It probably depends on the context from which this function is being called...
3. I also wonder if we can't simplify the Jinja2 handling, and also catch other weird values. For example, trying to `safe_load()` either `s = "{{ foo }}"` or `s = "@[foo]" both raise `yaml` exceptions (different ones, too!). Would it be a good idea to call `safe_load()` in a `try` and just return `s` if there's an error?

Considering the above, would something like this work?
```
try:
    r = yaml.safe_load(s)
except yaml.YAMLError:
    return s
return s if type(r) in [dict, list] else r
```

Reply:
It'll also handle type-ing `int` and `float` too! I like letting the YAMLError handle it,  I definitely agree with that!
Though in regards to 2, thats sort of why I put this up as a draft - while its lovely, consistent and clean here, it will absolutely reify everything. There may be even more things we want to keep as strings, so I wanted to make sure we accounted for all of them here. Though that brings up the point that, if we're only handling a small subset, we're not really just reifying anymore and should either clarify the subset in the function name or only call it when we're handling that subset

Reply:
I like the simplified `try/except` approach. Can we get away with removing the `strip` in the code given that we're now loading with YAML?

Reply:
Turns out the list is considered an `AttributeError` not `YAMLError`, so I made it catch both

Reply:
Sorry, what's the context where an `AttributeError` is raised? Is it related to this assertion in the unit tests?
```
assert nml_cfgobj.reify_scalar_str([1, 2, 3]) == [1, 2, 3]
```
If so, that's a `list` being passed in, not a `str`, so that shouldn't be a valid use case. We wouldn't want to catch an exception caused by that input, because the caller would be misusing the function.

Reply:
Oh I just processed that all wrong in my head, I've fixed it and added the fail on list as a case too